x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java

```diff
@@ -139,6 +140,7 @@ public static SystemIndexDescriptor getSystemIndexDescriptor() {
     private final NamedWriteableRegistry registry;
     private final Writeable.Reader<R> reader;
     private final BigArrays bigArrays;
+    private long maxResponseSize;
```

Comment:
I think this should be `volatile` to account for the live settings updates.

Reply:
Addressed in 219ddd5a5a698c497eb567253d59a8f7a666f8a3